scan_rr.go

```diff
@@ -846,6 +846,42 @@ func (rr *CSYNC) parse(c *zlexer, o string) *ParseError {
 	return nil
 }
 
+func (rr *ZONEMD) parse(c *zlexer, o string) *ParseError {
+	l, _ := c.Next()
+	i, e := strconv.ParseUint(l.token, 10, 32)
+	if e != nil || l.err {
+		return &ParseError{"", "bad ZONEMD Serial", l}
+	}
+	rr.Serial = uint32(i)
+
+	c.Next() // zBlank
+	l, _ = c.Next()
+	i, e1 := strconv.ParseUint(l.token, 10, 8)
+	if e1 != nil || l.err {
+		return &ParseError{"", "bad ZONEMD Scheme", l}
+	}
+	rr.Scheme = uint8(i)
+
+	c.Next() // zBlank
+	l, _ = c.Next()
+	if i, err := strconv.ParseUint(l.token, 10, 8); err != nil {
+		tokenUpper := strings.ToUpper(l.token)
+		i, ok := StringToAlgorithm[tokenUpper]
```

Comment:
This is wrong. The draft specifies only two supported algorithms SHA384 with value 1 and SHA512 with value 2; that doesn't match StringToAlgorithm. The "ZONEMD Hash Algorithms" will be a separate registry to the one used by DNSSEC.

Also, the draft doesn't specific the use of the mnemonic's here at all, instead only "unsigned decimal integer" are allowed. This should just be a parse error.

Reply:
you are right, we are removing that part